pandas/io/tests/test_gbq.py

```diff
@@ -430,10 +426,16 @@ def test_should_properly_handle_valid_integers(self):
         df = gbq.read_gbq(query, project_id=PROJECT_ID)
         tm.assert_frame_equal(df, DataFrame({'VALID_INTEGER': [3]}))
 
+    def test_should_properly_handle_valid_longs(self):
+        query = 'SELECT 4370565731996369294 as VALID_LONG'
+        df = gbq.read_gbq(query, project_id=PROJECT_ID)
+        tm.assert_frame_equal(
+            df, DataFrame({'VALID_LONG': [4370565731996369294]}))
+
     def test_should_properly_handle_null_integers(self):
         query = 'SELECT INTEGER(NULL) as NULL_INTEGER'
         df = gbq.read_gbq(query, project_id=PROJECT_ID)
-        tm.assert_frame_equal(df, DataFrame({'NULL_INTEGER': [np.nan]}))
+        tm.assert_frame_equal(df, DataFrame({'NULL_INTEGER': [None]}))
```

Comment:
np.NaN can not be stored in np.dtype(int)
http://pandas.pydata.org/pandas-docs/stable/gotchas.html#support-for-integer-na